lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,40 @@
+module Semian
+  module AtomicEnumSharedImplementation
```

Comment:
I would have thought that since the main detail of this file is implementing the `AtomicEnum` classes, whereas `AtomicEnumSharedImplementation` is just to facilitate sharing code, it should have been called atomic_enum. It's similar to `sliding_window_rb` in that it implements a class by that name. But alright.